From 9cacf2bcddbaf4513fe93f24a68ae5c944d17955 Mon Sep 17 00:00:00 2001
From: ningyu <ningyu@eswincomputing.com>
Date: Thu, 7 Aug 2025 10:41:23 +0800
Subject: [PATCH 475/540] feat:support S560A

Changelogs:
Implement S560A, need adapt bootspi flash, emmc, pmic and temperature sensor

Signed-off-by: ningyu <ningyu@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eic7702-d560a.dts | 83 +++++++++++++++++++++
 1 file changed, 83 insertions(+)
 create mode 100644 arch/riscv/boot/dts/eswin/eic7702-d560a.dts

diff --git a/arch/riscv/boot/dts/eswin/eic7702-d560a.dts b/arch/riscv/boot/dts/eswin/eic7702-d560a.dts
new file mode 100644
index 0000000000000..b5af295df905e
--- /dev/null
+++ b/arch/riscv/boot/dts/eswin/eic7702-d560a.dts
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree file for Eswin EIC7702 SoC.
+ *
+ * Copyright 2024, Beijing ESWIN Computing Technology Co., Ltd.. All rights reserved.
+ * SPDX-License-Identifier: GPL-2.0
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/dts-v1/;
+
+#include "eic7702-d560.dts"
+
+/ {
+	model = "ESWIN EIC7702 D560A";
+	compatible = "eswin,eic770x";
+};
+
+&bootspi {
+	/* spi flash PY25Q128LA */
+};
+
+&sdhci_emmc {
+	/* emmc YXSCG6PE-AK */
+};
+
+&d0_aon_i2c1 {
+	/* temp sensor(gx112) & pmic(ES5340) */
+	/delete-node/ tmp102@48;
+	/delete-node/ tps549d22@10;
+
+	es5340@f {
+		compatible = "einno,es5340";
+		reg = <0xf>;
+		eswin,regulator_default-microvolt=<1000000>;
+		eswin,regulator_label = "supply vdd1", "npu vdd1", "npu current1", "npu temperature1";
+		label = "npu_vdd";
+		regulators{
+			d0_npu_vcc:npu_svcc{
+				regulator-name="NPU_SVCC_D0";
+				regulator-min-microvolt=<700000>;
+				regulator-max-microvolt=<1100000>;
+				regulator-min-microamp=<20000000>;
+				regulator-max-microamp=<40000000>;
+				regulator-ov-protection-microvolt=<1100000>;
+			};
+		};
+	};
+};
+
+&d1_aon_i2c1 {
+	/* pmic(ES5340) */
+	/delete-node/ tps549d22@10;
+
+	es5340@f {
+		compatible = "einno,es5340";
+		reg = <0xf>;
+		eswin,regulator_default-microvolt=<1000000>;
+		eswin,regulator_label = "supply vdd1", "npu vdd1", "npu current1", "npu temperature1";
+		label = "npu_vdd";
+		regulators{
+			d1_npu_vcc:npu_svcc{
+				regulator-name="NPU_SVCC_D1";
+				regulator-min-microvolt=<700000>;
+				regulator-max-microvolt=<1100000>;
+				regulator-min-microamp=<20000000>;
+				regulator-max-microamp=<40000000>;
+				regulator-ov-protection-microvolt=<1100000>;
+			};
+		};
+	};
+};
\ No newline at end of file
-- 
2.53.0

